src/main/java/trackr/model/order/OrderNameContainsKeywordPredicate.java

```diff
@@ -1,30 +1,122 @@
 package trackr.model.order;
 
 import java.util.List;
+import java.util.Optional;
 import java.util.function.Predicate;
 
 import trackr.commons.util.StringUtil;
 
 /**
  * Tests that a {@code Order}'s {@code OrderName} matches any of the keywords given.
  */
-public class OrderNameContainsKeywordPredicate implements Predicate<Order> {
-    private final List<String> keywords;
+public class OrderNameContainsKeywordPredicate extends OrderDescriptor implements Predicate<Order> {
```

Comment:
Since now it does not only consider name, should the class be renamed accordingly